cobc/scanner.l

```diff
@@ -1283,11 +1461,46 @@ read_literal (const char mark, const char *type)
 			}
 			plex_buff[i] = (cob_u8_t)c;
 		}
+
+		if (ebcdic_flag) {
+			/* While scanning symbolic EBCDIC character: */
+			if (c == mark) {
+				if (ebcdic_flag == 2) {
+					cb_warning (COBC_WARN_FILLER,
+						    _("extraneous separator at end "
+						      "of symbolic EBCDIC characters"));
+				}
+				ebcdic_flag = 0;
+			} else if ('0' <= c && c <= '9') {
+				plex_buff[i++] = scan_ebcdic_char (c);
+				ebcdic_flag = 1;
+			} else if (c != ' ' && c != ',') {
+				cb_error (_("invalid character '%c' in sequence "
+					    "of symbolic EBCDIC characters"), c);
+				ebcdic_flag = 1;
+			} else if (ebcdic_flag == 2) {
+				cb_warning (COBC_WARN_FILLER,
+					    _("extraneous separator '%c' in sequence "
+					      "of symbolic EBCDIC characters"), c);
+				ebcdic_flag = 1;
+			} else {
+				ebcdic_flag = 2;
+			}
+			continue;
+		}
+
 		if (c == mark && (c = input ()) != (int)mark) {
 			if (c == '-') {
 				/* Free format continuation ("a"- 'b'- ) */
 				/* Hack it as concatenation */
 				unput ('&');
+			} else if (type[0] == '\0' && cb_ebcdic_literals &&
+				   '0' <= c && c <= '9') {
+				/* GCOS 7: symbolic EBCDIC character in
```

Comment:
As far as known: Is this GCOS only - or is there a similar extension with IBM COBOL or BS2000?

Reply:
I have not seen any mention of such symbolic characters in documentations of other dialects.